resources/day8/re_frame/trace/main.less

```diff
@@ -321,5 +329,55 @@
   .trace-details {
     cursor: pointer;
   }
-}
 
+  /* app-state data viewer
+     ========================================================================== */
+
+  .re-frame-trace--collection {
+    border-left: 1px solid @light-gray;
+    margin-left: 10px;
+    padding-left: 10px;
+    // background: rgba(0, 0, 0, 0.07);
+  }
+  .re-frame-trace--primative:not(:first-child) {
+    margin-left: 10px;
+    // color: @text-color-muted;
+  }
+  .re-frame-trace--number {
+  }
+  .re-frame-trace--boolean {
+    font-style: italic;
+  }
+  .re-frame-trace--string {
+    &:before {
+      color: @text-color-muted;
+      content: "\"";
+    }
+    &:after {
+      color: @text-color-muted;
+      content: "\"";
+    }
+  }
+  .re-frame-trace--nil {
+    text-decoration: line-through;
+    font-style: italic;
+  }
+  .re-frame-trace--keyword {
+    color: @light-purple;
+  }
+  .re-frame-trace--symbol {
+  }
+  .re-frame-trace--cljs-core-PersistentArrayMap,
```

Comment:
You don't want to be adding classes for concrete datatypes, you can check for `vector?`, `map?` instead. You could use the interface name if you liked though (IPersistentVector).